publication-commons/src/main/java/no/unit/nva/publication/service/impl/UpdateResourceService.java

```diff
@@ -188,18 +188,35 @@ private List<TransactWriteItem> createRefreshRelatedResourcesTransactions(Resour
         return Collections.emptyList();
     }
 
-    private Collection<TransactWriteItem> createResourceRelationTransactions(
+    private List<TransactWriteItem> createResourceRelationTransactions(
         Resource persistedResource,
         Resource resource) {
 
         if (anthologyRelationshipUnchanged(persistedResource, resource)) {
-            return List.of();
+            return Collections.emptyList();
         }
 
         var transactions = new ArrayList<TransactWriteItem>();
         deleteOldRelationTransaction(persistedResource).ifPresent(transactions::add);
         persistNewRelationTransaction(resource).ifPresent(transactions::add);
-        return transactions;
+        transactions.addAll(createRefreshAnthologyTransactions(persistedResource, resource));
+        return List.copyOf(transactions);
+    }
+
+    // TODO: After PK migration to be based on resource identifier only: update version using partial update without fetching resources
+    private List<TransactWriteItem> createRefreshAnthologyTransactions(Resource persistedResource, Resource resource) {
+        var oldAnthologyId = getAnthologyPublicationIdentifier(persistedResource);
+        var newAnthologyId = getAnthologyPublicationIdentifier(resource);
+
+        return Stream.of(oldAnthologyId, newAnthologyId)
+                   .filter(Optional::isPresent)
+                   .flatMap(Optional::stream)
+                   .distinct()
+                   .map(readResourceService::getResourceByIdentifier)
+                   .filter(Optional::isPresent)
+                   .map(Optional::get)
+                   .map(this::createPutTransaction)
+                   .toList();
     }
 
     private Optional<TransactWriteItem> deleteOldRelationTransaction(Resource resource) {
```

Comment:
not directly related to this PR, but wondering.. this code called like this
```
deleteOldRelationTransaction(persistedResource).ifPresent(transactions::add);
```
does it delete and create on every update?